include/ncengine/ecs/ComponentPool.h

```diff
@@ -101,7 +110,10 @@ class ComponentPool final : public ComponentPoolBase
 
         /** @brief Emplace a component attached to an entity. */
         template<class... Args>
-        auto Add(Entity entity, Args&&... args) -> T*;
+        auto Emplace(Entity entity, Args&&... args) -> T*;
+
+        /** @brief Insert a component attached to an entity. */
```

Comment:
I am not a big fan of using "Insert" here. Is its difference from `Emplace` just that you pass in an already existing component rather than the args to create one? If so, why not also call it `Emplace`? If you can't because `Emplace`'s parameter pack causes overload resolution to fail, I would choose a different name than insert.

Insert, to me and from what I've seen in the std, is different than emplace/push_back specifically because it requires a position in which the object is inserted into the container, and this method doesn't satisfy that convention.

Reply:
Hmm, interesting. I think there is a little more nuance here. The sequence containers, like `vector`/`deque`/`list`/`string`(I'm including here because of its API, even though it isn't technically this), have `insert`/`emplace` overloads that always take a position because they have something like `push_back`/`emplace_back` (lists are weird though, but who uses 'em?). The (unordered) associative containers, like `map`/`unordered_map`/`set`, all have `insert`/`emplace` that take a `value_type` only. They may also have overloads that also take a pos that acts as a hint.

I'm a little torn here, because this container acts as both, but I'm thinking `PushBack`/`EmplaceBack` convey the wrong message due to the item being moved to a staging pool initially, not the actual back of the pool. Can discuss further.

Reply:
Just wanted to add that I really like this comment. It kept me thinking while shoveling. If we had such a thing, I'd send you some corporate recognition coins.

Reply:
I'll buy a Nightcrawler Tee with my imaginary coins!
Agreed that since it is also partially an associative container that Insert works.